App/views.py

```diff
@@ -1,15 +1,22 @@
 from django.http import HttpResponse
 from django.shortcuts import render
+import math
 import json
 import requests
 import toolforge
 
 def main2(request):
     if(request.method=='POST'):
+        flag = 1
         usr = request.POST['username']
+        a = 1 if re.match("^[a-zA-Z0-9_ ]*$", s) else 0
+        if(!a):
```

Comment:
This doesn't look like valid Python. (Now that I look at it, all your ifs are in C-style. In Python you can just wrote `if reqest.method=='POST'` without the braces.)

Also, what's the point of the variable? You can always just inline it.

Reply:
Done.